src/components/core/geoChart/legends/test/PushpinCategoryLegend.spec.tsx

```diff
@@ -0,0 +1,40 @@
+// (C) 2020 GoodData Corporation
+import * as React from "react";
+import { shallow, ShallowWrapper } from "enzyme";
+import PushpinCategoryLegend, { IPushpinCategoryLegendProps } from "../PushpinCategoryLegend";
+import StaticLegend from "../../../../visualizations/chart/legend/StaticLegend";
+
+function createComponent(customProps: IPushpinCategoryLegendProps): ShallowWrapper {
+    const legendProps = {
+        ...customProps,
+    };
+    return shallow(<PushpinCategoryLegend {...legendProps} />);
+}
+
+describe("PushpinSizeLegend", () => {
+    it("should render component", () => {
+        const mockOnItemClick = jest.fn();
+        const segmentData = [
+            {
+                name: "General Goods",
+                legendIndex: 0,
+                color: "rgb(20,178,226)",
+                isVisible: true,
+            },
+            {
+                name: "Toy Store",
+                legendIndex: 1,
+                color: "rgb(0,193,141)",
+                isVisible: true,
+            },
+        ];
+        const props: IPushpinCategoryLegendProps = {
+            segmentData,
+            position: "top",
+            onItemClick: mockOnItemClick,
+        };
+        const wrapper = createComponent(props);
+        expect(wrapper.find(".viz-static-legend-wrap")).toHaveLength(1);
+        expect(wrapper.find(StaticLegend).length).toEqual(1);
```

Comment:
We have to assert number of segment items as well, it's 2 in this case.
And we have to have 'one enable and one disable items' case, then we assert the css which ones are enable/disable

Reply:
> We have to assert number of segment items as well, it's 2 in this case.
And we have to have 'one enable and one disable items' case, then we assert the css which ones are enable/disable

I think this should handle while writing unit test for StaticLegend.

Reply:
But we can assert it here.

Reply:
Or please check there is similar cases in `StaticLegend`.
If it is, we can skip this.

Reply:
added